### ISSUE TITLE
Factorio Server Manager crashes when using console in browser

### ISSUE THREAD

**Issue by Mattie112** [mroote/factorio-server-manager#76]
Chrome: Version 58.0.3029.110 (64-bit)
FSM: 0.6.0
Factorio: 0.15.9
OS: Ubuntu 16.04

Whenever I go to "Console" in the web UI my browser becomes unresponsive and the chrome debug console is spammed with these messages:
![image](https://cloud.githubusercontent.com/assets/662896/26099425/ef679548-3a2a-11e7-92c5-f349356fd9aa.png)

Whenever Chrome has enough of it and kills the webpage the server manager crashed with the following error:

> 2017/05/16 11:28:30 Factorio Server:    0.718 Hosting game at 0.0.0.0:34197
> 2017/05/16 11:28:30 Factorio Server:    0.718 Info ServerMultiplayerManager.cpp:664: mapTick(36001) changing state from(CreatingGame) to(InGame)
> 2017/05/16 11:28:30 Factorio Server:    0.740 Info RemoteCommandProcessor.cpp:122: Starting RCON interface at port 43081
> 2017/05/16 11:28:30 Rcon running on Factorio Server
> 2017/05/16 11:28:30 rcon session established on 0.0.0.0:43081
> 2017/05/16 11:28:30 Factorio Server:    0.740 Info RemoteCommandProcessor.cpp:230: New RCON connection from 127.0.0.1:34722
> 2017/05/16 11:28:30 Factorio server started on port: 34197
> panic: send on closed channel
> 
> goroutine 30 [running]:
> panic(0x7902c0, 0xc420136030)
>         /usr/local/go/src/runtime/panic.go:500 +0x1a1
> main.logSubscribe.func1(0xc42099ab10)
>         /home/mitch/prog/go/src/github.com/majormjr/factorio-server-manager/src/wsroutes.go:19 +0x1c2
> created by main.logSubscribe
>         /home/mitch/prog/go/src/github.com/majormjr/factorio-server-manager/src/wsroutes.go:21 +0x3f

Obviously resulting in my server being offline.

When I open the UI in Firefox (53.0.2) the messages are displayed in the web UI console window.
![image](https://cloud.githubusercontent.com/assets/662896/26099634/a124e16e-3a2b-11e7-9e7a-e7053e72bc9c.png)

This also results in the browser freezing, however when I force-close Firefox the  server-manager does NOT crash.

Please advice if you require any more info!

**Comment by Mattie112** [comment]
Correction: it does crash but only after one more action (loggin in again for example). Same error.

**Comment by sdobz** [comment]
I experienced this crash while attempting to start my server after being off for a day. 
```
2017/05/22 23:44:55 Starting server with command:  ../../factorio/bin/x64/factorio [--port 34197 --server-settings ../../factorio/config/server-settings.json --rcon-port 43081 --rcon-pass factorio_rcon --start-server ../../factorio/saves/_autosave1.zip]
2017/05/22 23:44:55 Factorio Server:    0.000 2017-05-22 23:44:55; Factorio 0.15.12 (build 29834, linux64, headless)
panic: send on closed channel

goroutine 91 [running]:
panic(0x7902c0, 0xc420aae250)
        /usr/local/go/src/runtime/panic.go:500 +0x1a1
main.logSubscribe.func1(0xc4202ca870)
        /home/mitch/prog/go/src/github.com/majormjr/factorio-server-manager/src/wsroutes.go:19 +0x1c2
created by main.logSubscribe
        /home/mitch/prog/go/src/github.com/majormjr/factorio-server-manager/src/wsroutes.go:21 +0x3f
```

Some environment information:
```
> ./run-manager.sh
[sudo] password for username:
2017/05/22 23:49:07 Loaded Factorio settings from ../../factorio/config/server-settings.json
2017/05/22 23:49:07 Created/Updated user: admin
Starting server on: 0.0.0.0:8080^C
> cat run-manager.sh
cd manager/factorio-server-manager

sudo -u factorio ./factorio-server-manager --dir ../../factorio
> lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 16.04.2 LTS
Release:        16.04
Codename:       xenial
```
I can't figure out which release of factorio-server-manager I have

**Comment by mroote** [comment]
Is the server crashing when the log gets spammed with messages for you also?  

It looks like @Mattie112's Factorio instance is getting some weird packets from their router and that is creating many log messages.  Too many log messages slows down the frontend it looks like and causes the browser to crash.  Then when the browser crashes the websocket connection gets closed abruptly and the manager then crashes.

I think there is two things to fix.  One is that the websocket connection closing needs to be handled more gracefully.  And two is to either improve the frontend rendering of log messages or maybe see if there is a way to rate limit log messages sent to the client.

**Comment by Mattie112** [comment]
Hi @mroote, yes I'm not sure if this problem is related to my (or a specific series) of routers but it is strange. The server works perfect and the correct ports are forwarded.

I also agree that there are 2 issues:
- The manager should not send messages to an already closed channel
- The manager should not spam the browser

If you want I can try to log in into Factorio and see if it crashes when the spamming starts or only when the browser/go channel is closed.

**Comment by xenoguy** [comment]
I've also experienced this issue.  to work around it i renamed the log file factoriofactorio-server-console.log.  maybe it has something to do with the size of the log?  it didn't seem huge or anything, only about 375kb.

on a busy server i imagine the logs get much bigger.

**Comment by dgw** [comment]
This issue would be less frustrating if FSM crashing didn't also seem to crash the running Factorio server too. (Gives some error about stdin, presumably because when FSM goes away the input stream becomes invalid or something. Haven't looked too deep.)

**Comment by knoxfighter** [comment]
This has been fixed with #178 and will added in the next release.
